src/AutoRest.CSharp/Common/Input/InputTypes/InputType.cs

```diff
@@ -27,4 +29,8 @@ internal InputType GetCollectionEquivalent(InputType inputType)
                 return inputType;
         }
     }
+
+    public string Name { get; internal set; } = Name;
+
+    public string? OriginalName { get; internal set; } = OriginalName;
```

Comment:
1. I don't think we should have any `set` for any `InputType`, which deviates from why we design it as `record`.
2. I think we should do something like:
```C#
internal abstract record InputType(string Name, bool IsNullable)
{
    public string CalculatedName => /*Some calculation*/;
    public string OriginalName => Name;
}
```
Because emitter doesn't do such calculation.